projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -2510,4 +2511,20 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
             this.regroupTrigger++;
         }
     }
+
+    /**
+     * @hidden @internal
+     */
+     public createRow(index: number, data?: any): RowType {
+        let row: RowType;
+
+        const dataIndex = this._getDataViewIndex(index);
+        const rec = data ?? this.dataView[dataIndex];
+
+
+        if (!row && rec) {
+            row = new IgxGridRow(this, index, rec);
```

Comment:
I don't think the IgxGridRow is a good representation of the PivotRow. It has a lot of setters and getters that should be overridden to not do anything or return always false if you look at the IgxPivotRowComponent. I think a new IgxPivotGridRow class should be introduced in this case.